omrmakefiles/rules.aix.mk

```diff
@@ -104,61 +103,58 @@ endif
 ### Shared Library Flags
 ###
 ifneq (,$(findstring shared,$(ARTIFACT_TYPE)))
+  # Export file
+  $(MODULE_NAME)_LINKER_EXPORT_SCRIPT := $(MODULE_NAME).exp
 
-# Export file
-$(MODULE_NAME)_LINKER_EXPORT_SCRIPT := $(MODULE_NAME).exp
+  define GENERATE_EXPORT_SCRIPT_COMMAND
+    sh $(top_srcdir)/omrmakefiles/generate-exports.sh xlc $(MODULE_NAME) $(EXPORT_FUNCTIONS_FILE) $($(MODULE_NAME)_LINKER_EXPORT_SCRIPT)
+  endef
 
-define GENERATE_EXPORT_SCRIPT_COMMAND
-sh $(top_srcdir)/omrmakefiles/generate-exports.sh xlc $(MODULE_NAME) $(EXPORT_FUNCTIONS_FILE) $($(MODULE_NAME)_LINKER_EXPORT_SCRIPT)
-endef
-
-ifneq (,$(findstring cxx_,$(ARTIFACT_TYPE)))
-  LINKTOOL:=$(CXXLINKSHARED)
-else
-  LINKTOOL:=$(CCLINKSHARED)
-endif
-ifeq (ld,$(LINKTOOL))
-  ifeq (1,$(OMR_ENV_DATA64))
-    GLOBAL_LDFLAGS+=-b64
+  ifneq (,$(findstring cxx_,$(ARTIFACT_TYPE)))
+    LINKTOOL:=$(CXXLINKSHARED)
   else
-    GLOBAL_LDFLAGS+=-b32
+    LINKTOOL:=$(CCLINKSHARED)
   endif
-  GLOBAL_LDFLAGS+=-G -bnoentry -bernotok -bnolibpath
-  GLOBAL_LDFLAGS+=-bmap:$(MODULE_NAME).map
-  GLOBAL_LDFLAGS+=-bE:$($(MODULE_NAME)_LINKER_EXPORT_SCRIPT)
-  GLOBAL_SHARED_LIBS+=c_r C_r m pthread 
-else
-  ifeq (1,$(OMR_ENV_DATA64))
-    GLOBAL_LDFLAGS+=-X64
+  ifeq (ld,$(LINKTOOL))
+    ifeq (1,$(OMR_ENV_DATA64))
+      GLOBAL_LDFLAGS+=-b64
+    else
+      GLOBAL_LDFLAGS+=-b32
+    endif
+    GLOBAL_LDFLAGS+=-G -bnoentry -bernotok -bnolibpath
+    GLOBAL_LDFLAGS+=-bmap:$(MODULE_NAME).map
+    GLOBAL_LDFLAGS+=-bE:$($(MODULE_NAME)_LINKER_EXPORT_SCRIPT)
+    GLOBAL_SHARED_LIBS+=c_r C_r m pthread
   else
-    GLOBAL_LDFLAGS+=-X32
+    ifeq (1,$(OMR_ENV_DATA64))
+      GLOBAL_LDFLAGS+=-X64
+    else
+      GLOBAL_LDFLAGS+=-X32
+    endif
+    GLOBAL_LDFLAGS+=-E $($(MODULE_NAME)_LINKER_EXPORT_SCRIPT)
+    GLOBAL_LDFLAGS+=-p 0 -brtl -G -bernotok -bnoentry -Wl,-bnolibpath
+    GLOBAL_SHARED_LIBS+=m
   endif
-  GLOBAL_LDFLAGS+=-E $($(MODULE_NAME)_LINKER_EXPORT_SCRIPT)
-  GLOBAL_LDFLAGS+=-p 0 -brtl -G -bernotok -bnoentry -Wl,-bnolibpath
-  GLOBAL_SHARED_LIBS+=m
-endif
-
-# Add map files to clean target
-define CLEAN_COMMAND
--$(RM) $(OBJECTS) $(MODULE_NAME).map
-endef
-
-# Shared library link command
-define LINK_C_SHARED_COMMAND
--$(RM) $@
-$(CCLINKSHARED) -o $@ $(OBJECTS) $(LDFLAGS) $(MODULE_LDFLAGS) $(GLOBAL_LDFLAGS)
-cp $@ $@.dbg
-endef
-
-define LINK_CXX_SHARED_COMMAND
--$(RM) $@
-$(CXXLINKSHARED) -o $@ $(OBJECTS) $(LDFLAGS) $(MODULE_LDFLAGS) $(GLOBAL_LDFLAGS)
-cp $@ $@.dbg
-endef
 
+  # Add map files to clean target
+  define CLEAN_COMMAND
```

Comment:
I think if this has its body indented, the whitespace will be copied when the command is used. I'm not sure about this.

Reply:
I did a small test with make 4.1: no leading whitespace as commands are echoed.